task/src/main/java/test_task/task/TaskLogic.java

```diff
@@ -2,17 +2,114 @@
 
 import com.amazonaws.services.lambda.runtime.Context;
 
+import cumulus_message_adapter.message_parser.AdapterLogger;
 import cumulus_message_adapter.message_parser.ITask;
 
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.sns.AmazonSNSClient;
+import com.amazonaws.services.sns.model.*;
+import com.amazonaws.regions.*;
+
+import com.google.gson.Gson;
+import java.util.Map;
+
+
 public class TaskLogic implements ITask
 {
     /**
-     * Sample business logic. Return string with task input.
+     * Get the SNS Topic Arn to publish to from the input JSON
+     * 
+     * @param input - input JSON
+     * @return Object containing the Topic Arn, null if not found
+     */
+    private Object GetTopicArn(String input)
+    {
+        Gson gson = new Gson();
+
+        Map inputMap = gson.fromJson(input, Map.class);
+
+        Object config = inputMap.get("config");
+
+        if(config != null)
+        {
+            Map configMap = gson.fromJson(gson.toJson(config), Map.class);
+            return configMap.get("topic_arn");
+        }
+
+        return null;
+    }
+
+    /**
+     * Get the Cumulus message to publish to SNS from the input JSON
+     * 
+     * @param input - input JSON
+     * @return Object containing the Cumulus message, null if not found
+     */
+    private Object GetCumulusMessage(String input)
+    {
+        Gson gson = new Gson();
+
+        Map inputMap = gson.fromJson(input, Map.class);
+
+        Object inputJson = inputMap.get("input");
+
+        if(inputJson != null)
+        {
+            Map messageInputMap = gson.fromJson(gson.toJson(inputJson), Map.class);
+            return messageInputMap.get("cumulus_message");
```

Comment:
I think cumulus_message is reserved word used by cumulus-message-adapter. You should use another key name.

Also we probably need a list of reserved words added to the documentation:

https://github.com/cumulus-nasa/cumulus-message-adapter/blob/bc564ae1f4627213356d7d198eab0d4f6485e4b3/message_adapter/message_adapter.py#L263